frontend/cache.go

```diff
@@ -0,0 +1,30 @@
+package main
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import "github.com/Azure/ARO-HCP/internal/api"
+
+type cache struct {
+	cluster map[string]api.HCPOpenShiftCluster
+}
+
+// NewCache returns a new cache.
+func NewCache() *cache {
+	return &cache{
+		cluster: make(map[string]api.HCPOpenShiftCluster),
+	}
+}
+
+func (c *cache) GetCluster(id string) (api.HCPOpenShiftCluster, bool) {
+	cluster, found := c.cluster[id]
+	return cluster, found
+}
+
+func (c *cache) SetCluster(id string, cluster api.HCPOpenShiftCluster) {
+	c.cluster[id] = cluster
+}
```

Comment:
Small suggestion, optional since I understand this code is temporary:
The cluster struct has the resource ID built in, so you technically don't need the `id` field.
```suggestion
func (c *cache) AddCluster(cluster api.HCPOpenShiftCluster) {
	c.cluster[cluster.Resource.ID] = cluster
}
```

Reply:
Since we don't have validation in place the ID isn't required to be passed in, so for now I'll keep the function how it is and keep the consistency with how the path is used

Reply:
Ok, makes sense.